hongit-api/src/main/kotlin/com/devfloor/hongit/api/article/application/ArticleService.kt

```diff
@@ -111,18 +154,47 @@ class ArticleService(
     }
 
     @Transactional(readOnly = true)
-    fun showAllByNickname(nickname: String): List<ArticleFeedResponse> {
-        val articles = userRepository.findByNicknameOrNull(nickname)
-            ?.let(articleRepository::findAllByAuthor)
+    fun showAllByNickname(nickname: String, page: Int, pageSize: Int): List<ArticleFeedResponse> {
+        val user: User = userRepository.findByNicknameOrNull(nickname)
             ?: EntityNotFoundException.notExistsNickname(User::class, nickname)
 
-        return articles
-            .filter { !it.anonymous }
+        val pageRequest: PageRequest = PageRequest.of(page, pageSize)
+        val articles = user
+            .let { articleRepository.findAllByAuthor(it, pageRequest) }
+
+        return articles.content.filter { !it.anonymous }
+            .map {
+                ArticleFeedResponse(
+                    article = it,
+                    articleOptions = articleOptionRepository.findAllByArticle(it),
+                    articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                    page = page,
+                    totalArticleCount = articleRepository.countAllByAuthor(user)
+                )
+            }
+    }
+
+    @Transactional(readOnly = true)
+    fun showAllByFavoritedUserId(userId: Long, page: Int, pageSize: Int): List<ArticleFeedResponse> {
+        val user: User = userRepository.findByIdOrNull(userId)
+            ?: EntityNotFoundException.notExistsId(User::class, userId)
+        val articleFavorites: List<ArticleFavorite> = articleFavoriteRepository.findAllByUser(user)
+        val pageRequest: PageRequest = PageRequest.of(page, pageSize)
+        val articles = articleFavorites
+            .let {
+                articleRepository.findAllByIdIn(it.map { it.id }, pageRequest)
+            }
+
+        return articles.content
             .map {
                 ArticleFeedResponse(
                     article = it,
                     articleOptions = articleOptionRepository.findAllByArticle(it),
                     articleFavorites = articleFavoriteRepository.findAllByArticle(it),
+                    page = page,
+                    totalArticleCount = articleRepository.countAllByIdIn(
+                        articleFavorites.map { it.id }
+                    )
                 )
             }
```

Comment:
```suggestion
        val articleFavoriteIds = articleFavoriteRepository.findAllByUser(user)
            .map { it.id }
        val articles = articleRepository.findAllByIdIn(articleFavoriteIds, PageRequest.of(page, pageSize))

        return articles.content
            .map {
                ArticleFeedResponse(
                    article = it,
                    articleOptions = articleOptionRepository.findAllByArticle(it),
                    articleFavorites = articleFavoriteRepository.findAllByArticle(it),
                    page = page,
                    totalArticleCount = articleRepository.countAllByIdIn(articleFavoriteIds)
                )
            }
```

이거 다시 전체적으로 보니깐 articleFavoriteIds만 사용하는 것 같은데 코드가 너무 불필요한 로컬 변수들이 많은 것 같아.
내가 한번 수정해봤는데 참고해봐바

Reply:
반영완료